### PR TITLE
v1.26.0 CSI and CPI chart image versions

### PR DESCRIPTION
**Pull request by dereknola**
https://github.com/rancher/image-mirror/pull/362 is a Prerequisite to this PR
#### Pull Request Checklist ####

- [X] Any new images or tags consumed by charts has been added [here](https://github.com/rancher/image-mirror)
- [X] Chart version has been incremented (if necessary)
- [X] That helm lint and pack run successfully on the chart.
- [x] Deployment of the chart has been tested and verified that it functions as expected.
- [x] Changes to scripting or CI config have been tested to the best of your ability

#### Types of Change ####
Added new Image version

    CSI  2.7.1, 3.0.1
    CPI 1.26.0 (note k8s v1.27.0 will pin to 1.26.0 as that is the latest release ATM)

Update image version:
   CSI 2.6.2 -> 2.6.3
   CSI 2.7.0 -> 2.7.1
   

#### After the PR is merged ####

Once the PR is merged, typically upon a new release, the necessary teams will be notified via Slack hook to perform the RKE2 Charts and RKE2 changes. Any developer working on this issue is not responsible for updating RKE2 Charts or RKE2.

### CODE REVIEW

**Review by briandowns — DISMISSED** [review]

File: charts/rancher-vsphere-csi/values.yaml
Comment (jiaqiluo) [inline charts/rancher-vsphere-csi/values.yaml:28]: I'd suggest not bumping tags in the "default" section for two reasons: 
1. they are always overwritten by the values from the versionOverrides section
2. afaik we did not bump the tags here each time when we bumped the same tags in the versionOverrides section, so there must be some mismatches between the components' tags, we never validate if it will work. 

@a-blender, please correct me if I say something wrong.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
I'd suggest bumping the tags under the versionOverrides section but not in the "default" section for two reasons: 
1. those values are always overwritten by the ones from the versionOverrides section
2. AFAIK, we do not bump them each time we bump the same tags in the versionOverrides section, there must be some mismatches between the components' tags, and we never validate if it will work. 

@a-blender, please correct me if I say something wrong.

**Comment by dereknola** [comment]
> 

Sounds good to me, I wasn't sure on the policy around the defaults.

**Review by jiaqiluo — APPROVED** [review]

**Review by a-blender — APPROVED** [review]
LGTM once both charts are tested on an rke cluster. Guide here https://github.com/rancher/vsphere-charts/blob/main/DEVELOPING.MD. I've found this particular test to be very helpful in the past, to verify CPI is working correctly

```
If you want to test that CPI is functioning properly, add the node.cloudprovider.kubernetes.io/uninitialized=true:NoSchedule taint to the cluster nodes before installing the chart. When you install CPI, the kubelet should remove the taints.
```

CSI also depends on CPI so imo it's necessary to specifically check this.

**Comment by dereknola** [comment]
@a-blender After several days of fighting vsphere, I have been unable to validate my changes and could use some help. I am not familiar with the rancher + vsphere setup process, so this has been quite the learning experience. I was able to provision a rke2 cluster with the `external` cloud provider via Rancher and then manually attempted to install the CSI/CPI charts. I filled out the `values.yaml` with the Fremont datacenter info, as described in the guide, and the chart seems to deploy just fine
```
root@rke2-try4-pool1-be5e6385-gtzr9:/home/ubuntu/vsphere-charts/charts# helm install -f ./rancher-vsphere-csi/values.yaml --kubeconfig=/etc/rancher/rke2/rke2.yaml csi ./rancher-vsphere-csi/
NAME: csi
LAST DEPLOYED: Tue May  9 17:24:31 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
```

However, the taint does not go away, and many pods are still pending. 
```
root@rke2-try4-pool1-be5e6385-gtzr9:/home/ubuntu/vsphere-charts/charts# kubectl get pods -A
NAMESPACE         NAME                                                     READY   STATUS             RESTARTS        AGE
calico-system     calico-kube-controllers-687bc88ddf-dbl52                 0/1     Pending            0               53m
calico-system     calico-node-4kghk                                        0/1     Running            0               53m
calico-system     calico-typha-689874cc98-5hw4f                            0/1     Pending            0               53m
cattle-system     cattle-cluster-agent-6c6c756b7c-jxglx                    0/1     Pending            0               54m
default           rancher-vsphere-cpi-cloud-controller-manager-cbwkz       0/1     CrashLoopBackOff   6 (2m11s ago)   8m21s
default           vsphere-csi-controller-7cd544bdc9-dxlcp                  0/5     Pending            0               4m52s
default           vsphere-csi-controller-7cd544bdc9-g56gr                  0/5     Pending            0               4m52s
default           vsphere-csi-controller-7cd544bdc9-skc29                  0/5     Pending            0               4m53s
```

Help validating this would be much appreciated!

**Review by brandond — APPROVED** [review]

File: charts/rancher-vsphere-cpi/values.yaml
Comment (brandond) [reply charts/rancher-vsphere-cpi/values.yaml:35]: compatibility is only "guaranteed" when the vsphere CPI and Kubernetes minors match, but in practice they do generally work properly on newer minors while we're waiting on upstream to get a new release out.

Comment (snasovich) [reply charts/rancher-vsphere-cpi/values.yaml:35]: I argue we should follow what is supported by upstream. Hypothetical, but if we have issues with CPI chart 1.26 on k8s 1.27 who is on the hook for investigating it and explaining to the customer?

Comment (brandond) [reply charts/rancher-vsphere-cpi/values.yaml:35]: historically we've just told users what I just said - not supported by upstream yet, but should work, and we'll update when upstream makes new images available.

**Comment by a-blender** [comment]
@dereknola Both charts deploy fine on a vsphere rke single node downstream cluster for me. CSI pods run successfully, CPI removed the taint from the node. I've checked the logs as well to make sure everything looks normal for the 1.26 images, I will merge this PR.

**Pull request merged**
